packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,318 @@
+
+        this.invokeOnChangeCallback(nextValue);
+    }
+
+    private updateValue(direction: number, e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) {
+        const delta = this.getDelta(direction, e);
+
+        // pretend we're incrementing from 0 if currValue isn't defined
+        const currValue = this.state.value || NumericStepper.VALUE_ZERO;
+        const nextValue = this.getAdjustedValue(currValue, delta);
+
+        this.setState({ shouldSelectAfterUpdate : true, value: nextValue });
+
+        this.invokeOnChangeCallback(nextValue);
+    }
+
+    private getAdjustedValue(value: string, delta: number) {
+        const { max, min } = this.props;
+
+        // truncate floating-point result to avoid precision issues when adding
+        // non-integer, binary-unfriendly deltas like 0.1
+        let nextValue = (!this.isValueNumeric(value))
+            ? 0
+            : parseFloat((parseFloat(value) + delta).toFixed(2));
+
+        nextValue = (min != null) ? Math.max(nextValue, min) : nextValue;
+        nextValue = (max != null) ? Math.min(nextValue, max) : nextValue;
+
+        return nextValue.toString();
+    }
+
+    private getDelta(direction: number, e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) {
+        if (e.shiftKey) {
+            return direction * this.props.majorStepSize;
+        } else if (e.altKey) {
+            return direction * this.props.minorStepSize;
+        } else {
+            return direction * this.props.stepSize;
+        }
+    }
+
+    private isValueNumeric(value: string) {
+        // checking if a string is numeric in Typescript is a big pain, because
+        // we can't simply toss a string parameter to isFinite. below is the
+        // essential approach that jQuery uses, which involves subtracting a
+        // parsed numeric value from the string representation of the value. we
+        // need to cast the value to the `any` type to allow this operation
+        // between dissimilar types.
+        return value != null && ((value as any) - parseFloat(value) + 1) >= 0;
+    }
+
+    private sortElements(inputGroup: JSX.Element, incrementButton: JSX.Element, decrementButton: JSX.Element) {
+        switch (this.props.buttonPosition) {
+            case "left":
+                return [decrementButton, incrementButton, inputGroup];
+            case "split":
+                return [decrementButton, inputGroup, incrementButton];
+            case "right":
+                return [inputGroup, decrementButton, incrementButton];
+            default:
+                // don't include the buttons.
+                return [inputGroup];
+        }
+    }
+
+    private getValueOrEmptyValue(props: INumericStepperProps) {
+        return (props.value != null)
+            ? props.value.toString()
+            : NumericStepper.VALUE_EMPTY;
+    }
+
+    private invokeOnChangeCallback(value: string) {
+        if (this.props.onChange) {
```

Comment:
Moved away from IControlledProps for now, and started having TS typing issues with safeInvoke. Need a hand to get that working again.